submissions/f3d0t/friends-app/app.js

```diff
@@ -0,0 +1,166 @@
+const DAY_NIGHT = document.querySelector(".day_night");
+const FRIENDS_COUNT = 60;
+const REQUEST_LINK = `https://randomuser.me/api/?results=${FRIENDS_COUNT}`;
+
+const fetchData = async (requestLink) => {
+	try {
+		const response = await fetch(requestLink);
+		const { results } = await response.json();
+		return results;
+	} catch (error) {
+		alert("HTTP-Error: " + error);
+	}
+};
+
+const initApp = (friendsDataArray) => {
+	const friends = new FriendsList(friendsDataArray, MAIN);
+	friends.renderCards();
+	bindEventListeners(friends);
+};
+
+const bindEventListeners = (friends) => {
+	FILTERS.addEventListener("input", ({ target }) => {
+		if (target.name == "search") {
+			friends.filterBySearch(target.value);
+		}
+		if (target.name == "gender") {
+			friends.filterByGender(target.id);
+			FILTERS.elements.search.value = "";
+			FILTERS.elements.name.forEach((radioButton) => {
+				if (radioButton.checked) {
+					friends.sortByName(radioButton.id);
+				}
+			});
+			FILTERS.elements.age.forEach((radioButton) => {
+				if (radioButton.checked) {
+					friends.sortByAge(radioButton.id);
+				}
+			});//this two forEach callbacks is needed to re-sort new arrays of cards, filtered by gender
+		}
+		if (target.name == "name") {
+			friends.sortByName(target.id);
+			FILTERS.elements.search.value = "";
+		}
+		if (target.name == "age") {
+			friends.sortByAge(target.id);
+			FILTERS.elements.search.value = "";
+		}
+	});
+	RESET_BUTTON.addEventListener("click", (e) => {
+		e.preventDefault();
+		FILTERS.querySelectorAll("input[type=radio]").forEach((radioButton) => {
+			radioButton.checked = false;
+		});
+		FILTERS.elements.search.value = "";
+		friends.resetFilters();
+	});
+	FILTERS.addEventListener("keydown", (event) => {
+		if (event.keyIdentifier == "U+000A" || event.keyIdentifier == "Enter" || event.keyCode == 13) {
+			event.preventDefault();
+		}
+	});
+};
+
+class FriendsList {
+	constructor(data, parentWrapper) {
+		this.allCards = data.map((friendData) => new FriendCard(friendData));
+		this.currentCards = [...this.allCards];
+		this.wrapper = parentWrapper;
+	}
+	renderCards(cardsArray = this.currentCards) {
+		this.wrapper.innerHTML = "";
+		this.wrapper.append(...cardsArray.map((card) => card.createCard()));
+	}
+	resetFilters() {
+		this.currentCards = [...this.allCards];
+		this.renderCards();
+	}
+	filterByGender(gender) {
+		if (gender == "male") {
+			this.currentCards = this.allCards.filter((friendCard) => friendCard["gender"] == "male");
+		}
+		if (gender == "female") {
+			this.currentCards = this.allCards.filter((friendCard) => friendCard["gender"] == "female");
+		}
+		if (gender == "all") {
+			this.currentCards = [...this.allCards];
+		}
+		this.renderCards();
+	}
+	filterBySearch(searchValue) {
+		const filteredArray = this.currentCards.filter((friendCard) => Object.values(friendCard).slice(1).join(" ").toLowerCase().indexOf(searchValue.toLowerCase()) > -1);
```

Comment:
First of all, you *really* should consider using prettier - it is a hell of one liner.
And second - what is going on here? :)

Reply:
Already using this, but the line width was set to 999 =(( fixing this

Reply:
This creates new, filtered array, which is implemented to not to modify sorted array, so you can search from sorted cards.

Reply:
I see. Please, split this logic into two steps - creation of array and then filtering them. This will improve readability

Reply:
it gets a one FriendCard, get the array from its values by Object.values, then slices by the first (because the first is gender, and I dont want to search by gender), then it joins array to a string by " " and then find that string contains searched value by indexOf method and LowerCasing them both =)

Reply:
Ok, in such a case, if someone will reorder keys in object, your slice will fail.
Please, extract `gender` using its key.
Second, would be cool to see some variable named `allValues` or `searchTarget`. Just for readability.